plutus-tx/src/PlutusTx/Builtins.hs

```diff
@@ -453,14 +457,14 @@ matchData
     -> (Integer -> r)
     -> (BuiltinByteString -> r)
     -> r
-matchData d constrCase mapCase listCase iCase bCase =
+matchData ~d ~constrCase ~mapCase ~listCase ~iCase ~bCase =
```

Comment:
Wow, I'm absolutely shocked by this - WTF?! I have no explanation.

Reply:
> I don't think that making function arguments lazy is going to do anything much?

Might prevent `matchData` from being inlined properly?

Reply:
> I don't think that making function arguments lazy is going to do anything much?

How so? Making formal parameters lazy is certainly going to make big differences. Making actual parameters lazy is useless.

Reply:
> this is all super suspicious and I want to know what's going on

Certainly suspicious, but so far I have no explanation.

Reply:
> Making formal parameters lazy is certainly going to make big differences

I don't agree since function application is strict. This should do literally nothing I think.